runners/direct-java/src/main/java/org/apache/beam/runners/direct/ParDoEvaluatorFactory.java

```diff
@@ -116,7 +116,7 @@ public void cleanup() throws Exception {
             mainOutputTag,
             sideOutputTags,
             stepContext,
-            fnManager.<InputT, OutputT>get(),
+            fnManager.<InputT, OutputT>get().getFn(),
```

Comment:
So, I think the status of this is that it makes sense to pass the `DoFnInvoker` here without pulling out the `DoFn`. This, I believe, will require adding `DoFnRunners.forInvoker` which may (or may not) pull out the `DoFn` in order to get its signature, or you could add `DoFnInvoker.getSignature()` which would be more flexible.